### PR TITLE
feat: Introduce Open Telemetry

### PR DESCRIPTION
**Pull request by BlankParticle**
## What does this PR do?

~Work In Progress PR for Open Telemtery~
~Uses Zipkin for now, still need to decide stuff~

Sets up basic tracing for Open Telemetry

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Chore (refactoring code, technical debt, workflow improvements)
- [ ] Enhancement (small improvements)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

### Required

- [ ] Read [Contributing Guide](https://github.com/un/inbox/blob/main/CONTRIBUTING.md)
- [ ] Self-reviewed my own code
- [ ] Tested my code in a local environment
- [ ] Commented on my code in hard-to-understand areas
- [ ] Checked for warnings, there are none
- [ ] Removed all `console.logs`
- [ ] Merged the latest changes from main onto my branch with `git pull origin main`
- [ ] My changes don't cause any responsiveness issues

### Appreciated

- [ ] If a UI change was made: Added a screen recording or screenshots to this PR
- [ ] Updated the UnInbox Docs if changes were necessary

### CODE REVIEW

**Comment by BlankParticle** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/un/inbox/487?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#487** <a href="https://app.graphite.dev/github/pr/un/inbox/487?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#486** <a href="https://app.graphite.dev/github/pr/un/inbox/486?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#484** <a href="https://app.graphite.dev/github/pr/un/inbox/484?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @BlankParticle and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: apps/mail-bridge/postal-routes/signature-middleware.ts [hunk 1]
```diff
@@ -1,32 +1,44 @@
...
-    return c.json({ message: 'Invalid signature' }, 401);
-  }
-
-  await next();
-});
+export const signatureMiddleware = createMiddleware<Ctx>(async (c, next) =>
+  c
+    .get('otel')
+    .tracer.startActiveSpan('Postal Signature Middleware', async (span) => {
+      if (c.req.method !== 'POST') {
```
Comment (McPizza0) [inline apps/mail-bridge/postal-routes/signature-middleware.ts:10]: why are we tracing postal signature?

**Review by McPizza0 — COMMENTED** [review]
@BlankParticle 
descriptions says
Work In Progress PR for Open Telemtery,
Uses Zipkin for now, still need to decide stuff

is this ready? have descisions been made?

**Comment by BlankParticle** [comment]
Yes, Otel base is ready, we need a exporter, but we can probably add that later if we dont have a place to store the data yet

**Comment by McPizza0** [comment]
add later, open issue for the outstanding work

**Review by McPizza0 — APPROVED** [review]

**Comment by BlankParticle** [comment]
### Merge activity

* **Jun 19, 8:51 AM EDT**: @BlankParticle merged this pull request with [Graphite](https://app.graphite.dev/github/pr/un/inbox/487).

**Pull request merged**